maintenance/wikia/deleteImageRevision.php

```diff
@@ -0,0 +1,164 @@
+<?php
+
+require_once( __DIR__ . '/../Maintenance.php' );
+
+class DeleteImageRevision extends Maintenance {
+
+	const PAGE_ID_OPTION = 'pageId';
+	const REVISION_ID_OPTION = 'revisionId';
+	const REASON_OPTION = 'reason';
+
+	private $db;
+
+	public function __construct() {
+		parent::__construct();
+		$this->addOption( self::PAGE_ID_OPTION, "image page id", true );
+		$this->addOption( self::REVISION_ID_OPTION, "revision id", true );
+		$this->addOption( self::REASON_OPTION, "deletion reason", true );
+		$this->mDescription = "Remove given revision of given image file";
+	}
+
+
+	public function execute() {
+		$pageId = intval( $this->getOption( self::PAGE_ID_OPTION ) );
+		$revisionId = intval( $this->getOption( self::REVISION_ID_OPTION ) );
+		$reason = $this->getOption( self::REASON_OPTION );
+
+		$title = Title::newFromID( $pageId );
+		if ( empty( $title ) ) {
+			\Wikia\Logger\WikiaLogger::instance()->error(
+				"can not find a file with page_id ${pageId}"
+			);
+
+			return;
+		}
+
+		$this->db = wfGetDB( DB_SLAVE );
+
+		$timestamp = $this->db->selectField(
+			[ 'revision' ],
+			'rev_timestamp',
+			[
+				'rev_page' => $title->getArticleID(),
+				'rev_id' => $revisionId,
+			],
+			__METHOD__
+		);
+
+		if ( $this->isOldImageRevision( $timestamp, $title->getDBkey() ) ) {
+			$this->removeOldRevision( $title, $timestamp, $revisionId, $pageId, $reason );
+		} else {
+			$this->removeLatestRevision( $title, $timestamp, $revisionId, $pageId, $reason );
+		}
+	}
+
+	private function removeLatestRevision( Title $title, string $revisionTimestamp, int $revisionId, int $pageId, string $reason ) {
+		$this->output(__METHOD__ . "\n");
+
+		if ( !$this->hasOldRevisions( $title->getDBkey() ) ) {
+			$file = wfLocalFile( $title );
+
+			// intentionally left as false to enforce removal of whole File by FileDeleteForm::doDelete when there is only one revision
+			$oldfile = false;
+			if ( !FileDeleteForm::doDelete( $title, $file, $oldfile, $reason, false )->isOK() ) {
+				\Wikia\Logger\WikiaLogger::instance()->error(
+					"deleting file was not successful",
+					[
+						'page_id' => $pageId,
+						'revision_id' => $revisionId
+					]
+				);
+			}
+		} else {
+			$this->revertToPreviousRevision( $title, $reason );
+			wfWaitForSlaves();
+			$this->removeOldRevision( $title, $revisionTimestamp, $revisionId, $pageId, $reason );
+		}
+	}
+
+	private function revertToPreviousRevision( Title $title, string $comment ) {
+		$this->output(__METHOD__ . "\n");
+
+		$archiveName = $this->db->selectField(
+			['oldimage'],
+			'oi_archive_name',
+			[
+				'oi_name' => $title->getDBkey()
+			],
+			__METHOD__,
+			['ORDER BY oi_timestamp desc']
+		);
+
+		//TODO: check on sandbox on test wiki if this method works as expected
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this "TODO" comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1135)